src/store/CanvasStore.ts

```diff
@@ -4,49 +4,110 @@ import { ref, type Ref } from 'vue'
 import IoConnection from '@/services/connection/IoConnection'
 import ShapeSerializer from '@/common/serializers/ShapeSerializer'
 import { useBoardStore } from './BoardStore'
+import { HistoryEventType, useHistoryStore } from './HistoryStore'
 
 export const useCanvasStore = defineStore('canvas', () => {
-  const boardStore = useBoardStore()
-
   const drawnShapes: Ref<Array<Shape>> = ref([])
   const currentlyDrawnShape: Ref<Shape | null> = ref(null)
   const canvasPosition = ref({ left: 0, top: 0 })
   const canvasOffset = ref({ x: 0, y: 0 })
 
-  function addDrawnShape(shape: Shape) {
+  function addDrawnShape(shape: Shape, record = true) {
     drawnShapes.value.push(shape)
-    IoConnection.emit('shape_create', {
-      board_id: boardStore.boardId,
-      shape: ShapeSerializer.toJson(shape)
-    })
+    if (record) recordShapeCreation(shape)
+    commitShapeCreation(shape)
   }
 
-  function removeDrawnShapeById(id: string) {
+  function removeDrawnShapeById(id: string, record = true) {
     const shapeIndex = drawnShapes.value.findIndex((shape) => shape.id == id)
     if (shapeIndex < 0) return
+    if (record) recordShapeDeletion(drawnShapes.value[shapeIndex])
     drawnShapes.value.splice(shapeIndex, 1)
-    IoConnection.emit('shape_delete', {
-      board_id: boardStore.boardId,
-      shape_id: id
-    })
```

Comment:
What happens to this?

Reply:
Moved to separate function: commitShapeDeletion